forc-plugins/forc-debug/src/server/handlers/handle_launch.rs

```diff
@@ -0,0 +1,202 @@
+use crate::server::{AdapterError, DapServer};
+use crate::types::Instruction;
+use forc_pkg::{self, BuildProfile, Built, BuiltPackage, PackageManifestFile};
+use forc_test::execute::TestExecutor;
+use forc_test::setup::TestSetup;
+use forc_test::BuiltTests;
+use std::{collections::HashMap, fs, sync::Arc};
+use sway_types::span::Position;
+
+impl DapServer {
+    /// Handles a `launch` request. Returns true if the server should continue running.
+    pub fn handle_launch(&mut self) -> Result<bool, AdapterError> {
+        // Build tests for the given path.
+        let (pkg_to_debug, test_setup) = self.build_tests()?;
+        let entries = pkg_to_debug.bytecode.entries.iter().filter_map(|entry| {
+            if let Some(test_entry) = entry.kind.test() {
+                return Some((entry, test_entry));
+            }
+            None
+        });
+
+        // Construct a TestExecutor for each test and store it
+        let executors: Vec<TestExecutor> = entries
+            .filter_map(|(entry, test_entry)| {
+                let offset = u32::try_from(entry.finalized.imm)
+                    .expect("test instruction offset out of range");
+                let name = entry.finalized.fn_name.clone();
+                if test_entry.file_path.as_path() != self.state.program_path.as_path() {
+                    return None;
+                }
+
+                Some(TestExecutor::new(
+                    &pkg_to_debug.bytecode.bytes,
+                    offset,
+                    test_setup.clone(),
+                    test_entry,
+                    name.clone(),
+                ))
+            })
+            .collect();
+        self.state.init_executors(executors);
+
+        // Start debugging
+        self.start_debugging_tests(false)
+    }
+
+    /// Builds the tests at the given [PathBuf] and stores the source maps.
+    pub(crate) fn build_tests(&mut self) -> Result<(BuiltPackage, TestSetup), AdapterError> {
+        if let Some(pkg) = &self.state.built_package {
+            if let Some(setup) = &self.state.test_setup {
+                return Ok((pkg.clone(), setup.clone()));
+            }
+        }
+
+        // 1. Build the packages
+        let manifest_file = forc_pkg::manifest::ManifestFile::from_dir(&self.state.program_path)
+            .map_err(|err| AdapterError::BuildFailed {
+                reason: format!("read manifest file: {:?}", err),
+            })?;
+        let pkg_manifest: PackageManifestFile =
+            manifest_file
+                .clone()
+                .try_into()
+                .map_err(|err: anyhow::Error| AdapterError::BuildFailed {
+                    reason: format!("package manifest: {:?}", err),
+                })?;
+        let member_manifests =
+            manifest_file
+                .member_manifests()
+                .map_err(|err| AdapterError::BuildFailed {
+                    reason: format!("member manifests: {:?}", err),
+                })?;
```

Comment:
I haven't looked into much detail in this part but if we already obtain a reference to the `member_manifests` to be able to build, we can also get the package manifest from that with the name of the package without needing to clone manifest file to cast into a package manifest file. `Assuming that we have the project name in place somewhere`.

```rust
let pkg_name = ...;
let pkg_manifest_file = &member_manifests[pkg_name];
```

Reply:
Addressed in https://github.com/FuelLabs/sway/pull/5625